src/utils/index.js

```diff
@@ -294,15 +294,52 @@ export const doesPlotContainCrop = plot =>
 
 export const getLifeStageRange = memoize((
   /** @type {farmhand.cropTimetable} */ cropTimetable
-) =>
-  [SEED, GROWING].reduce(
+) => {
+  return [SEED, GROWING].reduce(
     /**
      * @param {farmhand.cropLifeStage[]} acc
      */
-    (acc, stage) => acc.concat(Array(cropTimetable[stage]).fill(stage)),
+    (acc, stage) => {
+      if (Array.isArray(cropTimetable[stage])) {
+        return acc.concat(
+          cropTimetable[stage].reduce(
+            (acc2, value) => acc2.concat(Array(value).fill(stage)),
+            []
+          )
+        )
+      }
+
+      return acc.concat(Array(cropTimetable[stage]).fill(stage))
+    },
     []
   )
-)
+})
+
+/**
+ * @param {farmhand.crop} crop
+ * @returns {number}
+ */
+export const getGrowingPhase = memoize(crop => {
```

Comment:
If we're going to memoize this function, let's give it a `cacheSize` of `PURCHASEABLE_FIELD_SIZES.get(3).columns * PURCHASEABLE_FIELD_SIZES.get(3).rows`. Without that `cacheSize`, the memoization cache is liable to be busted before the Field is fully calculated, thus eliminating any performance gains.

Here's an example of using `cacheSize`: https://github.com/jeremyckahn/farmhand/blob/cfdab66a5347427e16310e8e65b0484b08a08a1b/src/utils/index.js#L456-L476

And here's `PURCHASEABLE_FIELD_SIZES`: https://github.com/jeremyckahn/farmhand/blob/cfdab66a5347427e16310e8e65b0484b08a08a1b/src/constants.js#L37-L43

Reply:
ah good call, though now i wonder... do you think we want this memoized? i was more or less just copying the other utils nearby here and didn't think much about it. i don't mind making this change if we do though

Reply:
I think there's value in it because this is a fairly hot code path. That said, there's a case to be made for dropping the memoization, as it does qualify as a premature optimization.

I vote for keeping the memoization (with appropriate cache configuration), but I'm okay with whichever way you'd like to go with this @lstebner.

Reply:
i'll add it, i thought maybe it didn't matter because the results to this method change almost every day, but i didn't analyze how often this might be called.